Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -704,6 +714,20 @@ Connection::AsyncReplyID Connection::sendWithAsyncReply(T&& message, C&& complet
     return { };
 }
 
+template<typename T>
+Ref<typename T::Promise> Connection::sendWithPromisedReply(T&& message, uint64_t destinationID, OptionSet<SendOption> sendOptions)
```

Comment:
If I read correctly, there is a discrepancy in how the code is written currently.
The idea is that the sendWithAsyncReply callback is:
* invoked in the function dispatcher of the connection
* If there's an error, the reply handler is dispatched in the main thread

The way you add this, the implication is that:
* you construct a promise probably bound to the current function dispatcher
* However, on success the promise is resolved in the function dispatcher of the connection
* If there's an error, the promise is resolved in the current function dispatcher

So only way this is going to be as-specified by the contract is if the sendWithPromisedReply is called on the function dispatcher of the connection. Currently there's no such requirement for the sendWithAsyncReply

Reply:
A promise itself isn't bound on any threads, it can be resolved anywhere: it's thread-safe.

It's the consumer of that promise that defines on which thread it requires the callback to run.

> If there's an error, the promise is resolved in the current function dispatcher
I have fixed this so that similar to sendWithAsyncReply, "If there's an error, the reply handler is dispatched in the main thread" , from there it will immediately run if the consumer's target queue is the main thread, or dispatched to the requested thread.

I guess it would be nice to avoid the dispatch if the consumer's target queue wasn't the main thread, but Connect has no way to know what the ultimate target queue is going to be.